### PR TITLE
Switch docker image source to Public ECR

### PR DESCRIPTION
**Pull request by samchungy**
To prevent us hitting rate limits when pulling images from Docker Hub we need to pull our base images from public ECR instead. This PR updates the templates and creates an autofix for the references in our Dockerfiles and Docker Compose files to pull from ECR.

Resolves https://github.com/seek-oss/skuba/issues/1332

### CODE REVIEW

File: template/lambda-sqs-worker-cdk/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # syntax=docker/dockerfile:1.10
 
-FROM --platform=<%- platformName %> node:20-alpine AS dev-deps
+FROM --platform=<%- platformName %> public.ecr.aws/docker/library/node:20-alpine AS dev-deps
```
Comment (AaronMoat) [inline template/lambda-sqs-worker-cdk/Dockerfile:3]: scope creep but should we be ditching --platform stuff given the docker warnings? @72636c

Comment (72636c) [reply template/lambda-sqs-worker-cdk/Dockerfile:3]: I'm comfortable with that 🚫🪟

Comment (samchungy) [reply template/lambda-sqs-worker-cdk/Dockerfile:3]: Donezo

File: src/cli/lint/internalLints/upgrade/patches/8.2.1/patchDockerImages.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import { inspect } from 'util';
+
+import fg from 'fast-glob';
+import { readFile, writeFile } from 'fs-extra';
+
+import type { PatchFunction, PatchReturnType } from '../..';
+import { log } from '../../../../../../utils/logging';
+
+const DOCKER_IMAGE_REGEX = /^(FROM\s?.*)(\s)(node|python)(:.*)/gm;
+const DOCKER_IMAGE_PLATFORM_REGEX = /^(FROM\s?.*)(--platform=[^\s]+) /gm;
```
Comment (AaronMoat) [inline src/cli/lint/internalLints/upgrade/patches/8.2.1/patchDockerImages.ts:10]: Hmm wait I think we only want to remove if it's a hardcoded value like arm64?

Comment (AaronMoat) [reply src/cli/lint/internalLints/upgrade/patches/8.2.1/patchDockerImages.ts:10]: Does this help: https://docs.docker.com/reference/build-checks/from-platform-flag-const-disallowed/

**Review by AaronMoat — APPROVED** [review]
Looks good tyvm!
